plugin/federation/federation.go

```diff
@@ -86,28 +86,55 @@ func (f *federation) MutateConfig(cfg *config.Config) error {
 }
 
 func (f *federation) InjectSourceEarly() *ast.Source {
-	input := `
-	scalar _Any
-	scalar _FieldSet
-	directive @requires(fields: _FieldSet!) on FIELD_DEFINITION
-	directive @provides(fields: _FieldSet!) on FIELD_DEFINITION
-	directive @extends on OBJECT | INTERFACE
-`
+	input := ``
+
 	// add version-specific changes on key directive, as well as adding the new directives for federation 2
 	if f.Version == 1 {
 		input += `
 	directive @key(fields: _FieldSet!) repeatable on OBJECT | INTERFACE
+	directive @requires(fields: _FieldSet!) on FIELD_DEFINITION
+	directive @provides(fields: _FieldSet!) on FIELD_DEFINITION
+	directive @extends on OBJECT | INTERFACE
 	directive @external on FIELD_DEFINITION
+	scalar _Any
```

Comment:
This line is still shared by both v1 and v2, correct?

Reply:
yeah but I thought it was cleaner to keep v1 and v2 definitions separate so they can be ordered (only common parts are `@extends` and `_Any`)